cirq/study/visualize.py

```diff
@@ -39,17 +37,18 @@ def plot_state_histogram(result: trial_result.TrialResult) -> np.ndarray:
     # This allows cirq to be usable without python3-tk.
     import matplotlib.pyplot as plt
 
-    num_qubits = len(result.measurements.keys())
+    assert set([value.shape[0] for value in result.measurements.values()]), \
+        'Same number of measurement must be made on all qubits.'
```

Comment:
Add a test that triggers the assertion (use `with pytest.raises(AssertionError, match='Same number of measurement must be made on all qubits'):` in the test).

Reply:
Guess this was a useless assertion, removing it. We call `repetitions=n` in the call to `sample()`, so I don't think different qubits will ever have different dimentions.